### PR TITLE
Document re-discovery of things

### PR DESCRIPTION
**Pull request by htreu**
Addresses #5526.

Signed-off-by: Henning Treu <henning.treu@telekom.de>

### CODE REVIEW

File: docs/documentation/development/bindings/discovery-services.md [hunk 1]
```diff
@@ -100,6 +100,12 @@ The following example implementation for `startScan` is taken from the `HueLight
     }
 ```
 
+### Re-Discovered Results and Things
+
+The `getThingUID` method of the discovery service should create a consistent UID every time the same thing gets discovered.
```
Comment (triller-telekom) [inline docs/documentation/development/bindings/discovery-services.md]: I would at least capitalize or maybe even emphasize `Thing` throughout the text.

File: docs/documentation/development/bindings/discovery-services.md [hunk 2]
```diff
@@ -100,6 +100,12 @@ The following example implementation for `startScan` is taken from the `HueLight
...
+This way existing discovery results and existing things with this UID will be updated with the properties from the current scan.
+With this dynamic discoveries (like UPnP or mDNS) can re-discover existing things and update communication properties like host names or TCP ports.
```
Comment (triller-telekom) [inline docs/documentation/development/bindings/discovery-services.md]: With this**,** dynamic...

Comment (htreu) [reply docs/documentation/development/bindings/discovery-services.md]: done

File: docs/documentation/development/bindings/discovery-services.md (hunk 1 above)
Comment (htreu) [reply docs/documentation/development/bindings/discovery-services.md]: There are a lot of other references to "thing" on the site. I would like to leave it like this for consistency reasons.

File: docs/documentation/development/bindings/discovery-services.md [hunk 3]
```diff
@@ -114,7 +120,14 @@ Normally, older discovery results already in the inbox are left untouched by a n
 
 ### Compare with existing results
 
-In some cases it might be of interest for a discovery service if the same device was discovered before or even a Thing already exists. Therefore, a discovery service may implement the ExtendedDiscoveryService interface and use the injected callback in order to access the DiscoveryResult or Thing for a given ThingUID, if it exists. 
+In some cases it might be of interest for a discovery service if the same device was discovered before or even a Thing already exists. Therefore, a discovery service may implement the ExtendedDiscoveryService interface and use the injected callback in order to access the DiscoveryResult or Thing for a given ThingUID, if it exists.
```
Comment (triller-telekom) [inline docs/documentation/development/bindings/discovery-services.md]: If you want to make it consistent, why did you write "Thing" with a capital "T" here? :) Also please add a line break after each sentence.

Comment (htreu) [reply docs/documentation/development/bindings/discovery-services.md]: Oh, you are right. Thanks.

File: docs/documentation/development/bindings/discovery-services.md [hunk 4]
```diff
@@ -114,7 +120,15 @@ Normally, older discovery results already in the inbox are left untouched by a n
 
 ### Compare with existing results
 
-In some cases it might be of interest for a discovery service if the same device was discovered before or even a Thing already exists. Therefore, a discovery service may implement the ExtendedDiscoveryService interface and use the injected callback in order to access the DiscoveryResult or Thing for a given ThingUID, if it exists. 
+In some cases it might be of interest for a discovery service if the same device was discovered before or even a thing already exists.
```
Comment (triller-telekom) [inline docs/documentation/development/bindings/discovery-services.md]: or even **that/if** a thing already exists

**Pull request merged**